File: public/algorithms/20.ValidParentheses/20.ValidParentheses.py
def isValid(self, s: str) -> bool:
        stack = []
        mapping = {')':'(', '}':'{', ']':'['}
        for c in s:
            if c in mapping.values():
                stack.append(c)
            else:
                if not stack or stack.pop() != mapping[c]:
                    return False
        return not stack

File: public/algorithms/20.ValidParentheses/20.ValidParentheses.js
var isValid = function (s) {
  const stack = [];
  const map = { ")": "(", "}": "{", "]": "[" };
  for (let c of s) {
    if (c === "(" || c === "{" || c === "[") stack.push(c);
    else {
      if (!stack.length || stack.pop() !== map[c]) return false;
    }
  }
  return stack.length === 0;
};
